common/utils/src/main/resources/error/error-classes.json

```diff
@@ -2807,6 +2844,24 @@
     ],
     "sqlState" : "07501"
   },
+  "NONEXISTENT_FIELD_NAME_INDEXING" : {
```

Comment:
Let's create an error class with sub-classes. BTW, cannot you re-use the existing one `FIELD_NOT_FOUND`?

Reply:
`FIELD_NOT_FOUND` is a good existing error to use. Removed duplicates and updated tests.